rfcs/react-components/convergence/event-handlers-event-type.md

```diff
@@ -32,71 +32,76 @@ We cannot just add the `Event` type to the existing callbacks as this is a break
 
 Use generic [`Event`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/v17/global.d.ts#L10) type and strongly-typed events in "data bag".
 
+### Implementation
+
+Define two helper types `EventData` and `EventHandler` to ensure consistency across all components:
+
+```ts
+export type EventData<Type extends string, TEvent> =
+  | { type: undefined; event: React.SyntheticEvent | Event | undefined }
```

Comment:
Why do we need undefined here in `event`? there's not a single case where we will not be providing an event, right?! I believe a generic signature without undefined suffices for this case

```suggestion
  | { type: undefined; event: React.SyntheticEvent | Event}
```